File: Semana 7/Ejercicios Excepciones (Cuaderno)/C++/03_excepcion_personalizada.cpp
#include <iostream>
using namespace std;

class MiExcepcion : public exception {
public:
    const char* what() const noexcept override {
        return "¡Excepción personalizada lanzada!";
    }
};

int main() {
    try {
        throw MiExcepcion();
    } catch (MiExcepcion& e) {
        cout << "Capturada: " << e.what() << endl;
    }
    return 0;
}

File: Semana 7/Ejercicios Excepciones (Cuaderno)/Python/03_excepcion_personalizada.py
class MiExcepcion(Exception):
    def __init__(self, mensaje="¡Excepción personalizada!"):
        super().__init__(mensaje)

try:
    raise MiExcepcion("Algo salió mal en mi app")
except MiExcepcion as e:
    print(f"Capturada: {e}")
